payment/key.go

```diff
@@ -35,25 +44,24 @@ func InitEncryptionKeys() {
 	copy(byteEncryptionKey[:], []byte(EncryptionKey))
 }
 
-// SetSecretKey decrypts the secret key from the database
-func (key *Key) SetSecretKey() error {
+// GetSecretKey decrypts the secret key from the database
+func (key *Key) GetSecretKey() (*string, error) {
```

Comment:
Much clearer.  Use the getter to get the secret key, check if secret key is nil.